client/my-sites/plans-grid/components/plan-type-selector/hooks/use-term-vise-max-dicounts.ts

```diff
@@ -0,0 +1,101 @@
+import {
+	PlanSlug,
+	URL_FRIENDLY_TERMS_MAPPING,
+	UrlFriendlyTermType,
+	getPlanMultipleTermsVariantSlugs,
+	getPlanSlugForTermVariant,
+	isMonthly,
+	isWpComPlan,
+	isWpcomEnterpriseGridPlan,
+} from '@automattic/calypso-products';
+import { UsePricingMetaForGridPlans } from 'calypso/my-sites/plans-grid/hooks/npm-ready/data-store/use-grid-plans';
+
+/**
+ * Calculate the maximum discount for each term for a given set of plans
+ * @param plans plans considered
+ * @param urlFriendlyTerms terms calculated for
+ * @param usePricingMetaForGridPlans price calculation hook
+ * @returns A map of term to maximum discount
+ */
+export default function useMaxDiscountsForPlanTerms(
```

Comment:
We changed the name here to `useMaxDiscountsForPlanTerms` (indeed better) but kept the filename `use-term-vise-max-dicounts` (it's also misspelled). We should probably filename too cc @jeyip

Reply:
Good catch. I'll fix it now 👍

Reply:
https://github.com/Automattic/wp-calypso/pull/85505